amr-wind/utilities/IOManager.cpp

```diff
@@ -142,6 +206,10 @@ void IOManager::write_plot_file()
         mesh.Geom(), m_sim.time().new_time(), istep, mesh.refRatio());
 
     write_info_file(plt_filename);
+
+#ifdef AMR_WIND_USE_ASCENT
+    ascent_pass(m_sim, m_plt_num_comp, m_plt_fields, m_plt_var_names);
+#endif
```

Comment:
This ties ascent output frequency to plot file write frequency. I would imagine that if in-situ visualization works we would most likely want a higher frequency of output for in-situ processed data. Instead of bolting on in-situ viz logic onto IOManager, Ascent capability should be a separate PostProcessing instance that can be activated independently of plot file output.

Reply:
Sounds good. Do you have a pointer to the best example to follow in creating a PostProcessing pass?

Reply:
The base class for post-processing is here: 

https://github.com/Exawind/amr-wind/blob/de3fd119f2e53b9360c6c9b6f9412521e4643497/amr-wind/utilities/PostProcessing.H#L20-L29

you can look at the `sampling` example which uses probes and planes to extract subset of data in NetCDF or text formats: 

https://github.com/Exawind/amr-wind/blob/de3fd119f2e53b9360c6c9b6f9412521e4643497/amr-wind/utilities/sampling/Sampling.H#L27-L40

An example of the input file:

https://github.com/Exawind/wind-energy/blob/28e8a9805653aaf3945a1d6ff379b51580545e8b/Pedersen_N02/AmrWind/abl.2.inp#L92-L107

Reply:
Excellent, thanks very much for your feedback!